implementations/rust/ockam/ockam_transport_tcp/src/router/mod.rs

```diff
@@ -51,10 +56,34 @@ impl TcpRouter {
         Ok(())
     }
 
-    async fn connect(&mut self, peer: String) -> Result<Address> {
+    async fn handle_unregister(&mut self, self_addr: Address) -> Result<()> {
+        trace!("TCP unregistration request: {}", &self_addr);
+        // TODO: Could have been much more efficient...
```

Comment:
I think this is just [`BTreeMap::retain`](https://doc.rust-lang.org/nightly/std/collections/struct.BTreeMap.html#method.retain), so could be `self.map.retain(|(_, v)| v != self_addr)`.

But it seems unlikely for the performance of this function to matter, unregister should be rare.

Reply:
@thomcc nice, I'll fix that